src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -482,15 +484,51 @@ class spell_warr_impending_victory : public SpellScript
 // 5246 - Intimidating Shout
 class spell_warr_intimidating_shout : public SpellScript
 {
-    void FilterTargets(std::list<WorldObject*>& unitList)
+    void FilterTargets(std::list<WorldObject*>& unitList) const
     {
         unitList.remove(GetExplTargetWorldObject());
     }
 
+    void ClearTargets(std::list<WorldObject*>& unitList) const
+    {
+        // This is used in effect 3, which is an AOE Root effect.
+        // This doesn't seem to be a thing anymore, so we clear the targets list here.
+        unitList.clear();
+    }
+
     void Register() override
     {
-        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_warr_intimidating_shout::FilterTargets, EFFECT_1, TARGET_UNIT_SRC_AREA_ENEMY);
         OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_warr_intimidating_shout::FilterTargets, EFFECT_2, TARGET_UNIT_SRC_AREA_ENEMY);
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_warr_intimidating_shout::ClearTargets, EFFECT_3, TARGET_UNIT_SRC_AREA_ENEMY);
+    }
+};
+
+// 316594 - Intimidating Shout (Menace Talent, knock back -> root)
+class spell_warr_intimidating_shout_menace_knock_back : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_INTIMIDATING_SHOUT_MENACE_AOE });
+    }
+    void FilterTargets(std::list<WorldObject*>& unitList) const
+    {
+        unitList.remove(GetExplTargetWorldObject());
+    }
+
+    void HandleRoot(SpellEffIndex /*effIndex*/) const
+    {
+        CastSpellExtraArgs const args(GetSpell());
+        GetCaster()->m_Events.AddEventAtOffset([caster = GetCaster(), targetGUID = GetHitUnit()->GetGUID(), castArgs = args]()
```

Comment:
Should we be making a copy of CastSpellExtraArgs here? I assume it's to avoid lifetime issues since the lambda is delayed, but I'm unsure whether CastSpellExtraArgs is temporary or not.

Reply:
You can, however, castArgs = CastSpellExtraArgs(GetSpell()) directly on the lambda, there's no need for it to be outside.